app/src/MainController.cpp

```diff
@@ -397,20 +395,21 @@ void MainController::poll_events() {
         m_spotLightColor[2] = std::max(m_spotLightColor[2] - 0.02f, 0.0f);
     }
 
-    if (platform->key(engine::platform::KeyId::KEY_T).is_down() && !m_starKeyPressed) {
+    if (platform->key(engine::platform::KeyId::KEY_T).is_down() && !m_starEnabled) {
```

Comment:
Ovaj kod i dalje nije `thread-safe`, naime:
glavna nit može pročitati `m_starEnabled = false` i ući u if-granu, i startovati nit sa `std::thread(&MainController::alter_star, this).detach();` i odmah nakon te naredbe biti sklonjena sa CPU-a od strane kernela, zatim novo napravljena nit se može izvršiti do kraja, i onda se nastaviti izvršavanje glavne niti koja će postaviti `m_StarEnabled=true` i nakon toga okidanje ovog događaja više neće biti moguće. 

prilikom provere m_starEnabled koristite `compare_exchange` i onda neka nova nit samo vrati vrednost kada završi,
`m_starEnabled=true` na liniji 400 je onda višak